tests/Integration/Behaviour/Features/Context/ShopFeatureContext.php

```diff
@@ -65,16 +72,133 @@ public function shopWithNameExists(string $reference, string $shopName)
     /**
      * @Given single shop context is loaded
      */
-    public function singleShopContextIsLoaded()
+    public function singleShopContextIsLoaded(): void
     {
         Shop::setContext(Shop::CONTEXT_SHOP, Configuration::get('PS_SHOP_DEFAULT'));
     }
 
     /**
      * @Given multiple shop context is loaded
      */
-    public function multipleShopContextIsLoaded()
+    public function multipleShopContextIsLoaded(): void
     {
         Shop::setContext(Shop::CONTEXT_ALL);
     }
+
+    /**
+     * @Given I add a shop group :reference with name :groupName and color :color
+     *
+     * @param string $reference
+     * @param string $groupName
+     * @param string $color
+     */
+    public function addShopGroup(string $reference, string $groupName, string $color)
+    {
+        $shopGroup = new ShopGroup();
+        $shopGroup->name = $groupName;
+        $shopGroup->color = $color;
+        $shopGroup->active = true;
+        if (!$shopGroup->add()) {
+            throw new RuntimeException(sprintf('Could not create shop group: %s', Db::getInstance()->getMsgError()));
+        }
+
+        SharedStorage::getStorage()->set($reference, $shopGroup);
+    }
+
+    /**
+     * @Given I add a shop :reference with name :shopName and color :color for the group :shopGroupName
+     *
+     * @param string $reference
+     * @param string $shopName
+     * @param string $shopGroupName
+     */
+    public function addShop(string $reference, string $shopName, string $color, string $shopGroupName): void
+    {
+        $shop = new Shop();
+        $shop->active = true;
+        $shop->id_shop_group = ShopGroup::getIdByName($shopGroupName);
+        $shop->id_category = 2;
+        $shop->theme_name = _THEME_NAME_;
+        $shop->name = $shopName;
+        $shop->color = $color;
+        if (!$shop->add()) {
+            throw new RuntimeException(sprintf('Could not create shop: %s', Db::getInstance()->getMsgError()));
+        }
+        $shop->setTheme();
+
+        SharedStorage::getStorage()->set($reference, $shop);
+    }
+
+    /**
+     * @When I search for shops with the term :searchTerm I should get the following results:
+     *
+     * @param string $searchTerm
+     * @param TableNode $table
+     */
+    public function assertFoundShops(string $searchTerm, TableNode $table): void
```

Comment:
FYI if each table row represents FoundShop DTO then you can cast it using `@Transform`
like this https://github.com/PrestaShop/PrestaShop/pull/22244/files#diff-82fec04e223dab8f56f4604efd062e83a9058da2851559d55c88ca0f902d20c6R63-R102

Reply:
hmm, interesting, I'll have a look 👍

Reply:
Ok, I started doing it, but the problem in my case is that FoundShop is an existing shop, it takes a shop ID and a group shop ID in the constructor, and since IDs might change depending on the tests executed before in the same feature suite, I don't want to rely on them for anything...

Reply:
But I didn't know about this @transform keyword, I'll definitely use it when possible.

Reply:
> it takes a shop ID and a group shop ID in the constructor, and since IDs might change depending on the tests executed before in the same feature

In theory each step should be independent from each other (we are far from it though :sweat: ).
Using references to reflect ids would work if Background contains only "exists" assertion (when shop is created only if it does not exist yet) whilst now for each scenario it is creating a new shop while keeping same reference (that seems to be wrong :sweat:). 
E.g. if you add another scenario, then your shops would be duplicated by the background and you would expect to find multiple shops with same names and same references - it doesn't seem right.

I would suggest simply to move following lines from background to scenario
```
    Given I add a shop group "shopGroup2" with name "test_second_shop_group" and color "green"
    And I add a shop "shop2" with name "test_second_shop" and color "red" for the group "test_second_shop_group"
    And I add a shop "shop3" with name "test_third_shop" and color "blue" for the group "test_second_shop_group"
```
Then there won't be any problem adding another independant scenario and also you could take advantage of transformers by using references for shopId and groupId.
Im not very confident with this as actually somehow we should encourage using background :sweat:  so I ping @jolelievre to check this out, at least to see the problem with backgrounds